.github/workflows/node.js.yml

```diff
@@ -14,6 +14,10 @@ jobs:
     - name: Checkout
       uses: actions/checkout@v2
 
+    - name: Install xsltproc
+      run: |
+        sudo apt-get install xsltproc
+
     - name: Cache
```

Comment:
That can be done, although I'm not sure if restoring across Node.js versions is the best idea. Sometimes, switching to a new Node.js major requires a full re-installation or some binary packages may not install properly.

Reply:
Also, I don't see a point of including both `${{ runner.os }}-node-${{ matrix.node-version }}-npm-lock-` and `${{ runner.os }}-node-${{ matrix.node-version }}-` - they have the same variable paths so if one matches, the other one should as well.

Reply:
...although, to that point, the official docs also show a similar example:
https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows
Do you see a reason to include all of them?

Reply:
I think the restore keys work like a fallback chain for always restoring something at all and then updating it. Speeds up builds when using version matrix. Never had issues with it but not 100% sure if it's not possible to cause issues.

Reply:
I don't think it will cause issues, it just looks redundant. There should be no situation where the cache entry for `${{ runner.os }}-node-${{ matrix.node-version }}-npm-lock-` for given `os` & `node-version` values is matched but for `${{ runner.os }}-node-${{ matrix.node-version }}-` for those same values it does not match.